packages/@webex/plugin-meetings/src/multistream/remoteMediaManager.ts

```diff
@@ -0,0 +1,893 @@
+      // pane id doesn't exist, so nothing to do
+      LoggerProxy.logger.log(
+        `RemoteMediaManager#removeMemberVideoPane --> removeMemberVideoPane() called for a non-existent paneId: ${paneId} (pane not found in this.media.video.memberPanes)`
+      );
+
+      return Promise.resolve();
+    }
+
+    const remoteMedia = this.media.video.memberPanes[paneId];
+
+    const receiveSlot = remoteMedia.getUnderlyingReceiveSlot();
+
+    if (receiveSlot) {
+      this.receiveSlotManager.releaseSlot(receiveSlot);
+
+      const index = this.slots.video.receiverSelected.indexOf(receiveSlot);
+
+      if (index >= 0) {
+        this.slots.video.receiverSelected.splice(index, 1);
+      }
+    }
+    remoteMedia.stop();
+
+    delete this.media.video.memberPanes[paneId];
+    delete this.currentLayout.memberVideoPanes?.[paneId];
+
+    return Promise.resolve();
+  }
+
+  /**
+   * Pins an active speaker remote media object to the given CSI value. From that moment
+   * onwards the remote media will only play audio/video from that specific CSI until
+   * unpinActiveSpeakerVideoPane() is called or current layout is changed.
+   *
+   * @param {RemoteMedia} remoteMedia remote media object reference
+   * @param {CSI} csi CSI value to pin to, if undefined, then current CSI value is used
+   */
+  public pinActiveSpeakerVideoPane(remoteMedia: RemoteMedia, csi?: CSI): void {
+    const remoteMediaGroup = Object.values(this.media.video.activeSpeakerGroups).find((group) =>
+      group.includes(remoteMedia, false)
+    );
+
+    if (!remoteMediaGroup) {
+      throw new Error('remoteMedia not found among the active speaker remote media groups');
+    }
+
+    remoteMediaGroup.pin(remoteMedia, csi);
+  }
+
+  /**
+   * Unpins a remote media object from the fixed CSI value it was pinned to.
+   *
+   * @param {RemoteMedia} remoteMedia remote media object reference
+   */
+  public unpinActiveSpeakerVideoPane(remoteMedia: RemoteMedia) {
+    const remoteMediaGroup = Object.values(this.media.video.activeSpeakerGroups).find((group) =>
+      group.getPinnedRemoteMedia().includes(remoteMedia)
+    );
+
+    if (!remoteMediaGroup) {
+      throw new Error(
+        'remoteMedia not found among the pinned remote media from any active speaker group'
+      );
+    }
+
+    remoteMediaGroup.unpin(remoteMedia);
+  }
+
+  /**
+   * Returns true if a given remote media object belongs to an active speaker group and has been pinned.
+   * Throws an error if the remote media object doesn't belong to any active speaker remote media group.
+   *
+   * @param {RemoteMedia} remoteMedia remote media object
+   * @returns {boolean}
+   */
+  public isPinned(remoteMedia: RemoteMedia) {
+    const remoteMediaGroup = Object.values(this.media.video.activeSpeakerGroups).find((group) =>
+      group.includes(remoteMedia, true)
+    );
+
+    if (!remoteMediaGroup) {
+      throw new Error(
+        'remoteMedia not found among the pinned remote media from any active speaker group'
```

Comment:
Error message incorrect, remoteMedia is not found among the remote media from any active speaker group, pinned or unpinned.

Reply:
fixed